lib/FusionInventory/Agent/Task/Inventory/Virtualization/Xen.pm

```diff
@@ -79,18 +80,30 @@ sub  _getVirtualMachines {
     my $line  = <$handle>;
 
     my @machines;
-    while (my $line = <$handle>) {
+    while ($line = <$handle>) {
         chomp $line;
+        next if $line =~ /^\s*$/;
         my ($name, $vmid, $memory, $vcpu, $status);
         my @fields = split(' ', $line);
         if (@fields == 4) {
-                ($name, $memory, $vcpu) = @fields;
-                $status = 'off';
+            ($name, $memory, $vcpu) = @fields;
+            $status = 'off';
         } else {
-                ($name, $vmid, $memory, $vcpu, $status) = @fields;
-                $status =~ s/-//g;
-                $status = $status ? $status_list{$status} : 'off';
-               next if $vmid == 0;
+            if ($line =~ /^(.*\S) \s+ (\d+) \s+ (\d+) \s+ (\d+) \s+ ([a-z-]{5,6}) \s/x) {
+                ($name, $vmid, $memory, $vcpu, $status) = ($1, $2, $3, $4, $5);
+            } elsif ($params{logger}) {
+                # message in log to easily detect matching errors
+                my $message = '_getVirtualMachines(): unrecognized output';
+                if ($params{command}) {
```

Comment:
Indeed. However, I'd expect test files won't trigger such a warning, and if they does, the test should pass an empty command parameter to avoid the issue.

My point is that the this kind of corner case should be handled in testing code, not in actual agent code, except when there is no other way. The main agent code is run by every user every day, whereas test are only run sometimes by developers, it does not make sense to have everyone pay the price for it.